solr/core/src/java/org/apache/solr/update/SolrIndexSplitter.java

```diff
@@ -130,9 +131,26 @@ public SolrIndexSplitter(SplitIndexCommand cmd) {
     }
     routeFieldName = cmd.routeFieldName;
     if (routeFieldName == null) {
-      field = searcher.getSchema().getUniqueKeyField();
+      // To support routing child documents, use the root field if it exists (which would be populated with unique field),
+      // otherwise use the unique key field
+      field = searcher.getSchema().getFieldOrNull(IndexSchema.ROOT_FIELD_NAME);
+      if(field == null) {
+        field = searcher.getSchema().getUniqueKeyField();
+      }
     } else  {
-      field = searcher.getSchema().getField(routeFieldName);
+      SchemaField uniqueField = searcher.getSchema().getUniqueKeyField();
+      if (uniqueField.getName().equals(routeFieldName)) {
+        // Explicitly routing based on unique field
+        // To support routing child documents, use the root field if it exists (which would be populated with unique field),
+        // otherwise use the unique key field
+        field = searcher.getSchema().getFieldOrNull(IndexSchema.ROOT_FIELD_NAME);
+        if (field == null) {
+          field = searcher.getSchema().getUniqueKeyField();
+        }
+      } else {
+        // Custom routing
+        field = searcher.getSchema().getField(routeFieldName);
+      }
```

Comment:
I don't think we should modify the logic for when a routerFieldName is provided; should simply be unsupported.  In my experience, it's rare to use that.  I'd prefer the code read simpler if you remove the support; you duplicated code for it.

Reply:
It kinda seems like the opposite. The routeFieldName seems to be pre-populated by the zk collection state (not from split caller (api doesn't support setting afaict)). Doesn't that mean it'll usually be set?

https://github.com/apache/lucene-solr/blob/2d63a9d1208bbf950135b90496268b0a40e119b5/solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java#L118-L135

Reply:
Notice it's conditional based on the router having a "field" in the JSON.  It's rare to use that.

Also if we don't support something (child docs + whatever), we ought to throw an exception instead of do the wrong thing.  See `org.apache.solr.schema.IndexSchema#isUsableForChildDocs`